lib/ClimaCoreTempestRemap/src/netcdf.jl

```diff
@@ -0,0 +1,213 @@
+
+import ClimaCore: slab, column
+
+"""
+    write_space_cgll!(nc::NCDataset, space::Spaces.AbstractSpace)
+
+Create the dimensions for `space` in the `nc` dataset.
+
+If a compatible dimension already exists, it will be reused.
+"""
+function write_space_cgll!(nc::NCDataset, space::Spaces.SpectralElementSpace2D)
+    nodes = Spaces.unique_nodes(space)
+    ncol = length(nodes)
+
+    if haskey(nc, "lon")
+        # dimension already exists: check correct size
+        if size(nc["lon"]) != (ncol,)
+            error("incompatible horizontal dimension already exists")
+        end
+        return nothing
+    end
+
+    # dimensions
+    defDim(nc, "ncol", ncol)
+
+    # variables
+    ## lon
+    lon = defVar(nc, "lon", Float64, ("ncol",))
+    lon.attrib["units"] = "degrees_east"
+    lon.attrib["axis"] = "X"
+    lon.attrib["long_name"] = "longitude"
+    lon.attrib["standard_name"] = "longitude"
+
+    ## lat
+    lat = defVar(nc, "lat", Float64, ("ncol",))
+    lat.attrib["units"] = "degrees_north"
+    lat.attrib["axis"] = "Y"
+    lat.attrib["long_name"] = "latitude"
+    lat.attrib["standard_name"] = "latitude"
+
+    coords = Spaces.coordinates_data(space)
+
+    for (col, ((i, j), e)) in enumerate(nodes)
+        coord = slab(coords, e)[i, j]
+        lon[col] = coord.long
+        lat[col] = coord.lat
+    end
+    return nothing
+end
+
+function write_space_cgll!(
+    nc::NCDataset,
+    space::Spaces.CenterFiniteDifferenceSpace,
+)
+    nlevels = Spaces.nlevels(space)
+
+    if haskey(nc, "z")
+        if size(nc["z"]) != (nlevels,)
+            error("incompatible vertical dimension already exists")
+        end
+        return nothing
```

Comment:
Why do we escape here if `haskey(nc, "z")`?

Reply:
the dimension is already defined.

I was thinking of making this function part of `write_var_cgll!`, so it could be called multiple times.

Reply:
Hmm, ok. The way we do it in TC.jl is partitioned into 2 parts: "adding fields" (writing meta info), and writing fields. At first glance it looks like we could partition things the same here. The reason I suggest this is because of [this issue](https://github.com/Alexander-Barth/NCDatasets.jl/issues/135). At least for EDMF, repeatedly writing meta info was dominating the cost of the entire simulation because of the quadratic behavior with the number of variables. Obviously this is fine for only a few variables, but it may be something to consider.

Reply:
That seems like a good idea.

Reply:
I've updated the interface to be more explicit.